content/rest-api/index.textile

```diff
@@ -461,6 +469,251 @@ bc[json]. [{
   data: <optional clientData provided by the client>
 }]
 
+h2. Push
+
+h3(#post-device-registration). Register a device for receiving push notifications
+
+Register or update a device's details, including the information necessary to deliver push notifications to it.
+
+h6. POST rest.ably.io/push/deviceRegistrations
+
+The request body contains device details and is an object of the form:
+
+bc[json]. {
+  id: <string>,
+  clientId: <optional client identifier to associate with registration>,
+  platform: <ios, android, browser>,
+  formFactor: <mobile, tablet, desktop, other>,
+  metadata: <object with additional data>,
+  push: {
+    transportType: <gcm, fcm, apns, web>,
+    metadata: <object with additional data, like transport-specific recipient address>
+  }
+}
+
+bc[sh]. curl -X POST https://rest.ably.io/push/deviceRegistrations \
+ -u "{{API_KEY}}"
+ -H "Content-Type: application/json" \
+ --data \
+ '
+ {
+  "id": "01ARZ3NDEKTSV4RRFFQ69G5FAV",
+  "platform": "ios",
+  "formFactor": "mobile",
+  "push": {
+    "transportType": "apns",
+    "metadata": {
+      "deviceToken": "740f4707bebcf74f9b7c25d48e3358945f6aa01da5ddb387462c7eaf61bb78ad"
+    }
+  }
+}
+'
+
+h5. Parameters
+
+None
+
+h5. Options
+
+- Content-Type := @application/json@, @application/x-msgpack@ or @application/x-www-form-urlencoded@
+- Accept := not applicable
+- Auth required := yes ("basic":#basic-authentication or "token":#token-authentication)
+
+A successful request returns the just-registered device details. It includes a @updateToken@, which is a token that can be used to "authenticate":#token-authentication afterwards to update an existing device registration (for example, for updating a GCM or FCM registration token).
+
+A unsuccessful request returns an error.
+
+h3(#post-channel-subscription). Subscribe to a channel
+
+Subscribe either a single device or all devices associated with a client ID to receive push notifications from messages sent to a channel.
+
+h6. POST rest.ably.io/push/channelSubscriptions
+
+The request body contains subscription details and is an object of the form:
+
+bc[json]. {
+  deviceId: <optional, string, cannot be set when clientId is set>,
```

Comment:
Perhaps change this to `<optional, string, must be set if `clientId` is empty, cannot be set when `clientId` is set>`